tck/features/expressions/pattern/Pattern2.feature

```diff
@@ -28,5 +28,211 @@
 
 #encoding: utf-8
 
-Feature: Pattern2 - Number of Pattern Matches
+Feature: Pattern2 - Pattern Comprehension
 
+  Scenario: [1] Returning a pattern comprehension
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)
+      CREATE (a)-[:T]->(:B),
+             (a)-[:T]->(:C)
+      """
+    When executing query:
+      """
+      MATCH (n)
+      RETURN [p = (n)-->() | p] AS ps
+      """
+    Then the result should be, in any order:
+      | ps                                     |
+      | [<(:A)-[:T]->(:C)>, <(:A)-[:T]->(:B)>] |
+      | []                                     |
+      | []                                     |
+    And no side effects
+
+  Scenario: [2] Returning a pattern comprehension with label predicate
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B), (c:C), (d:D)
+      CREATE (a)-[:T]->(b),
+             (a)-[:T]->(c),
+             (a)-[:T]->(d)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      RETURN [p = (n)-->(:B) | p]
+      """
+    Then the result should be, in any order:
+      | [p = (n)-->(:B) \| p] |
+      | [<(:A)-[:T]->(:B)>]   |
+    And no side effects
+
+  Scenario: [3] Returning a pattern comprehension with bound nodes
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
+      CREATE (a)-[:T]->(b)
+      """
+    When executing query:
+      """
+      MATCH (a:A), (b:B)
+      RETURN [p = (a)-[*]->(b) | p] AS paths
+      """
+    Then the result should be, in any order:
+      | paths               |
+      | [<(:A)-[:T]->(:B)>] |
+    And no side effects
+
+  Scenario: [4] Introducing new node variable in pattern comprehension
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a), (b {name: 'val'})
+      CREATE (a)-[:T]->(b)
+      """
+    When executing query:
+      """
+      MATCH (n)
+      RETURN [(n)-[:T]->(b) | b.name] AS list
+      """
+    Then the result should be, in any order:
+      | list    |
+      | ['val'] |
+      | []      |
```

Comment:
I think it would add an interesting level to this scenario if we also added another node `c` which is connected like `(b)-[:T]->(c)` and which does _not_ have a `name` property. That would mean the result adds a third row with a `[null]` value, illustrating that the result list size varies not with the presence of the property, but with the cardinality of the pattern.

Reply:
I was initially confused because I thought `[]` corresponded to the `a` node not having a name, but it corresponds to the `b` node having no outgoing rels. My suggestion may help another reader.

Reply:
I added you suggestion. The cardinality beyond 0 and 1 can only be tested correctly by using `size` since the outcome of a pattern comprehension is indeterministic with regards to the order of the elements in the list, as far as I understand things. It is a problem in scenario [1], which I now have changed to not be indeterministic.